cinema/views.py

```diff
@@ -87,6 +115,8 @@
         )
     )
     serializer_class = MovieSessionSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
 
     def get_queryset(self):
         date = self.request.query_params.get("date")
```

Comment:
Ensure that the date format in the query parameter matches the expected format. Consider adding error handling for invalid date formats to prevent potential runtime errors.